docs/migration-guides/external-providers-primitives-v2.mdx

```diff
@@ -0,0 +1,213 @@
+description: Migrate to the primitives interface in external providers
+
+---
+
+# Migrate provider interfaces from backend.run to primitives
+
+## Motivation and context
+
+The evolution of quantum algorithms and scaling of quantum processing units (QPUs) motivated the
+introduction of the Qiskit [primitives interface](/guides/primitives) to provide a higher-level abstraction for
+accessing QPUs. This interface is optimized for two core tasks in quantum algorithm development: expectation value estimation (Estimator) and
+circuit sampling (Sampler).
+
+After a time of co-existence of the `backend.run` model and primitives model, where the primitives
+evolved to the updated V2 primitives, the `backend.run` model was finally deprecated in favor of
+these newer interfaces to streamline quantum workload execution.
+
+The existing collection of migration guides allow users to [transition to the runtime provider](/migration-guides/qiskit-runtime) and
+[update their user code to the new primitives V2 interface](/migration-guides/v2-primitives) in time for the upcoming removal of `backend.run`.
+To complement that, this guide focuses on supporting external providers (both existing and upcoming)
+in the implementation their own flavor of the primitives V2 interface (Estimator or Sampler) to integrate seamlessly into the
+standard Qiskit workflow.
+
+These implementations can be used to wrap any custom provider hardware access function
+(for example: `execute_workload(QPU)` or `resource.access()`), or local simulator,
+as long as the final inputs and outputs conform to the established contract from the primitive interfaces.
+
+## If your provider already implemented `backend.run`
+
+The Qiskit SDK offers ready-to-go wrappers for `backend.run` that can be easily adapted to a custom primitives workflow
+through subclassing, these are the [`BackendEstimatorV2`](/api/qiskit/qiskit.primitives.BackendEstimatorV2) and [`BackendSamplerV2`.](/api/qiskit/qiskit.primitives.BackendSamplerV2) The inputs
+to the primitives should follow the Primitive Unified Bloc (PUB) syntax specified in the V2 primitives interface. See the [Overview of PUBs section](/guides/primitive-input-output#pubs) in the Primitive inputs and outputs guide for details.
+
+An advantage of this strategy is that the wrapper can handle the input/output manipulation, so knowledge of the PUB data
+model is not required. However, this might result in a sub-optimal runtime, which could be refined through a fully
+custom primitives implementation.
+
+The snippets below show how to create a custom Estimator instance following the strategy described above.
+The process would be analogous for a custom Sampler, modifying the base class to `BackendSamplerV2`.
+
+``` python
+from qiskit.primitives import BackendEstimatorV2
+
+class CustomEstimator(BackendEstimatorV2):
+    """Estimator primitive for custom provider."""
+
+    _backend: CustomProviderResource
```

Comment:
This line is overwriting the type of the `self._backend` attribute to a custom type, because `BackendEstimatorV2` is written to only accept IBM types. It's a bit of a technicality, Python doesn't enforce these types so the code would run without it, but it feels a bit hacky not to add it. Let me come up with a comment to clarify this.

Reply:
Added in [d1a12e8](https://github.com/Qiskit/documentation/pull/2496/commits/d1a12e80cc756f9fb65e11ea81de83bb41b3f36a)